iceoryx_posh/source/mepoo/shm_safe_unmanaged_chunk.cpp

```diff
@@ -41,13 +41,14 @@ ShmSafeUnmanagedChunk::ShmSafeUnmanagedChunk(mepoo::SharedChunk chunk) noexcept
     // this is only necessary if it's not an empty chunk
     if (chunk)
     {
-        rp::RelativePointer<mepoo::ChunkManagement> ptr = chunk.release();
+        rp::RelativePointer<mepoo::ChunkManagement> ptr{chunk.release()};
         auto id = ptr.getId();
         auto offset = ptr.getOffset();
         cxx::Ensures(id <= rp::RelativePointerData::ID_RANGE && "RelativePointer id must fit into id type!");
         cxx::Ensures(offset <= rp::RelativePointerData::OFFSET_RANGE
                      && "RelativePointer offset must fit into offset type!");
-        m_chunkManagement = rp::RelativePointerData(static_cast<rp::RelativePointerData::id_t>(id), offset);
+        /// @todo #1196 Unify types to uint64_t
```

Comment:
What is supposed to happen here in the todo?

Reply:
`RelativePointerData` defines it as `using id_t = uint16_t` and `BaseRelativePointer` as `uint64_t`. This should be addressed in a follow-up PR.

Reply:
I'd rather make a `uint16_t` instead of a `uint64_t`. We have to store this together with the offset as 64 bit in the shared memory. When the id is an `uin16_t` there is less to check for in the `RelativePointerData` :)